controllers/chat/ChatController.js

```diff
@@ -426,6 +426,135 @@ module.exports = {
       return ErrorResponse.e400(res, error.message);
     }
   },
+  initChatSourceAnonymous: async (req, res) => {
+    const {
+      targetUser,
+      message,
+      anon_user_info_color_code,
+      anon_user_info_color_name,
+      anon_user_info_emoji_code,
+      anon_user_info_emoji_name
+    } = req.body;
+
+    let members = [targetUser];
+    if (!members.includes(req.userId)) members.push(req.userId);
+
+    const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    try {
+      /**
+       * @type {import('stream-chat').OwnUserResponse}
+       */
+      const user = {
+        name: `Anonymous ${anon_user_info_emoji_name}`,
+        id: req.userId,
+        image: '',
+        username: `Anonymous ${anon_user_info_emoji_name}`
+      };
+      await client.connectUser(user, req.token);
+
+      if (client.user.name !== `Anonymous ${anon_user_info_emoji_name}`) {
+        await client.upsertUser({id: req.userId, name: `Anonymous ${anon_user_info_emoji_name}`});
+      }
+
+      const channel = client.channel('messaging', {members});
+      const createdChannel = await channel.create();
+
+      await channel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name
+        }
+      });
+
+      // Default mock chat in case of channel is blocked
+      let chat = {
+        message: {
+          id: uuidv4(),
+          cid: createdChannel?.channel?.id,
+          text: message,
+          message,
+          user: {
+            id: req.userId,
+            name: `Anonymous ${anon_user_info_emoji_name}`,
+            image: '',
+            username: `Anonymous ${anon_user_info_emoji_name}`
+          },
+          created_at: new Date(),
+          updated_at: new Date()
+        }
+      };
+
+      if (!createdChannel?.channel?.is_channel_blocked) {
+        chat = await channel.sendMessage({
+          user_id: req.userId,
+          text: message,
+          ...req.body
+        });
+      }
+
+      const targets = members.filter((member) => member !== req.userId);
+      await targets.map(async (target) => {
+        const exist = await ChatAnonUserInfo.count({
+          where: {
+            channel_id: channel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target
+          }
+        });
+
+        if (!exist) {
+          await ChatAnonUserInfo.create({
+            channel_id: channel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target,
+            anon_user_info_color_code,
+            anon_user_info_color_name,
+            anon_user_info_emoji_code,
+            anon_user_info_emoji_name
+          });
+        }
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You're using `Array.prototype.map` but not actually using the resulting array. If you just want to perform an action for each item in the array, use `Array.prototype.forEach` instead. Also, you are performing asynchronous operations inside the loop. It would be more efficient to use `Promise.all` to execute these operations in parallel.

```diff
- await targets.map(async (target) => {
+ const promises = targets.map((target) => {
    // ... rest of the code ...
  });
+ await Promise.all(promises);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->